internal/app/basin_provider.go

```diff
@@ -10,9 +10,10 @@ import (
 
 // DealInfo represents information about a deal.
 type DealInfo struct {
-	ID           uint64
-	SelectorPath string
```

Comment:
so there won't any way to know the selector path for the data?

Reply:
i don't see a reason for having that. but we could add if that makes sense

Reply:
a deal could have many uploaded files in it, correct? ie, user uploads files `a`, `b`, `c`. web3.storage gives us back a cid for each, but _may_ put them all in one, two, or three deals... we don't know. so in order to get the data for cid `a` out of whatever deal it's in, we need a way to identify it in the CAR file. or am i complicating it?

Reply:
yes, thats my understanding too based on this [doc](https://docs.filecoin.io/basics/how-retrieval-works/basic-retrieval).

i think, for hot layer retrieval `selectorPath` is not required because we have CIDs for each file that we upload. but if the data is purged from hot storage, then the only way to get it would be from storage provider, right? which would require a selector path. 

i also added a comment on the other PR but it is related: https://github.com/tablelandnetwork/basin-provider/pull/12#discussion_r1380310389

Reply:
will the CLI be able to do the lassie things for the user? ie, can we obfuscate where the data is coming from (hot or cold)?

Reply:
not sure how lassie works, but i'd assume we can probably import lassie libraries into the CLI and offer similar features. I can take a look at that

Reply:
>Not sure what kind of magic is going on there, but looks like it's definitely possible to retrieve from Filecoin only with CID.

interesting! we should test it somehow but I thought `lassie` is only able to fetch data using just the CID because w3s is pinning it. and it will continue pinning until we keeping paying them. this is why i am not sure what happens to the CID if it not pinned by w3s anymore

Reply:
good point! i thought lassie was only a Filecoin retriever, but could be the case that they are using IPFS

Reply:
asked on Filecion slack channel

- Does lassie fetches content from some Filecoin storage provider or from the IPFS node that pinned the file?
> Both
- How can I make sure lassie fetches only from Filecoin?
> You can pass a flag to lassie to limit either the protocols or providers to allow
- And how from only the CID lassie can figure out the deal, and the path inside the deal where the content is?
> lassie looks up the the CID in [#ipni](https://filecoinproject.slack.com/archives/C02T827T9N0) - the network indexers - which store a database of CID->provider.
For filecoin providers, the provider locally stores metadata to help it serve the CIDs it advertises - like where they are inside the deal

So, looks like with only the CID the client can retrieve a file, we can make use of lassie as a library if needed.

Also, if we want to list all deals of a file, we can create another command specific to that.

Reply:
this is great, so makes sense to remove the path and ID 👍

Reply:
Sweet